pkg/client/cache/delta_fifo.go

```diff
@@ -452,6 +460,16 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will send a sync event for each item
+func (f *DeltaFIFO) Resync() error {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+	if err := f.replaceLocked(f.listLocked()); err != nil {
```

Comment:
Resync can do deletion detection with the replace impl because it is looking at its queue of deltas and comparing to what is in the local cache.  That state should really only happen if a watch notification isn't sent, which is exactly the case that TestUpdate is testing.  In normal operation  I'm not sure that is realistic though.  I'm fine changing this back to queueActionLocked.